patches/chrome-browser-ui-browser.cc.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/chrome/browser/ui/browser.cc b/chrome/browser/ui/browser.cc
+index 6e0ce3191b697c1f45ae9fd5b73679f50c131cd5..b46546f5a8c511ff524902e3d13c69528de27ee4 100644
+--- a/chrome/browser/ui/browser.cc
++++ b/chrome/browser/ui/browser.cc
+@@ -555,6 +555,9 @@ Browser::~Browser() {
+   // because the InstantController destructor depends on this profile.
+   instant_controller_.reset();
+ 
++  // Clear browsing data on exit, if needed.
++  content::BraveClearBrowsingData::ClearOnExit(profile_);
```

Comment:
I think we can do this with BrowserListObserver::OnBrowserRemoved or chrome::NOTIFICATION_BROWSER_CLOSED instead of patching

Reply:
I need to look at this more though because this is safe to do here for a persistent profile. Seems like a potential problem on shutdown because the browsing data deletion is async and the profile could be destroyed before it finishes. The shutdown crash @bsclifton is seeing has a strange stack trace, but I think it probably stems from that issue

Reply:
I'm not sure this is completely safe though. It seems like a potential problem on shutdown with async deletion of data going on during shutdown. Can you verify that `BrowsingDataRemoverImpl::Notify` is called on shutdown?

Reply:
I know something similar is called for guest/incognito, but those are in_memory profiles with minimal data on disk

Reply:
@bridiver, that was my biggest concern as well. I have a browsertest (above) BraveClearDataOnExitTest.VerifyRemovalCompletion that calls content::BrowsingDataRemover::RemoveAndReply instead of just Remove. The test then waits and in this case BrowsingDataRemoverImpl::Notify does get called. In regular browser exit, though, the UI thread does go away before all the clean up is done so Notify doesn't get called (usually 2 tasks would still be left when I look at it in the debugger), but I am not sure that means that the clean up doesn't complete on another thread.

Reply:
Hey, Browser destructor doesn't look like a proper place to do this, because Browser represents only a single window. The user may open quite a few number of windows for every opened profile. So its better to peek at `browser_shutdown.cc` to catch the shutdown start or to `profile_impl.cc` to hook into profile destruction.

Reply:
That's an interesting point @iefremov. On Windows, as a user, I think I would expect the data to be cleared as I close all windows associated with the given profile because that's when they would go away from the task bar. On a Mac, though, if there is a visual indicator that the app is still running, even though all windows are closed, I would not expect the data to be cleared. What was Muon's behavior in this regard, @bsclifton?

Reply:
@mkarolin clear on exit (in Muon) will only clear once the app is actually quitting (versus the window count). I think it's fair to say the expected use-case for this PR is application exit